app/assets/javascripts/discourse/app/components/text-field.js

```diff
@@ -53,12 +53,13 @@ export default TextField.extend({
     next(() => this.onChange(this.value));
   },
 
-  @discourseComputed
-  dir() {
+  get dir() {
     if (this.siteSettings.support_mixed_text_direction) {
-      let val = this.value;
-      if (val) {
-        return isRTL(val) ? "rtl" : "ltr";
+      const val = this.get("value");
```

Comment:
```suggestion
      const val = this.value;
```

Reply:
`this.value` is just an EmberObject property - it's not `@tracked`. Therefore we need to use `get` in order to keep things reactive.

(this change would make the new test fail)

Reply:
We're using native getters with EmberObjects already, what's different here?

Reply:
`this.value` isn't tracked, and it isn't a glimmer component `this.args` object. Unfortunately that means we gotta use `get()` so that the Ember Autotracking system knows that this getter is dependent on it.

(Ultimately this stems from the fact that `@ember/component/text-field` is a classic component using the old reactivity model)